### PR TITLE
Add backend for forgot password feature

### LINKED ISSUE #126: Forgot Password Feature
**Is your feature request related to a problem? Please describe.**
There should exist a forgot password feature so that if any user forgets their password they can change set a new password.
 
**Describe the solution you'd like**
There should be a forgot password feature.

**Additional context**
If this feature gets approved then i would love to work on it!

### PR DESCRIPTION
**Pull request by carpecodeum**
### Description
Added the backend functionality for forgot password feature. Added nodemailer package for mailer purposes.
Fixes #126 

### Type of Change:
**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality)

Screenshot 2020-02-24 at 6 04 49 PM

[senz-forgotpassword-testing.zip](https://github.com/scorelab/senz/files/4254517/senz-forgotpassword-testing.zip)

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged

### CODE REVIEW

**Comment by YashMeh** [comment]
Kindly add a video depicting functionality.

**Comment by carpecodeum** [comment]
Sure!

**Comment by carpecodeum** [comment]
@YashMeh I have uploaded a zip file please review that!

**Comment by YashMeh** [comment]
@carpecodeum attach a video url instead of a GIF.
Also, resolve the conflicts.

**Comment by carpecodeum** [comment]
> @carpecodeum attach a video url instead of a GIF.
> Also, resolve the conflicts.

@YashMeh please have a look now.

File: senz-web/backend/.env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ENV="test"
+EMAIL_PASSWORD = "adi9891304474"
```
Comment (YashMeh) [inline senz-web/backend/.env]: **Remove** the .env file from the PR and reset your password for security reasons.

File: senz-web/backend/routes/userRoutes.js [hunk 2]
```diff
@@ -49,7 +51,7 @@ const getSignature = username => {
  */
 
 //Register a new user
-router.post("/register", function(req, res) {
+router.post("/register", function (req, res) {
```
Comment (YashMeh) [inline senz-web/backend/routes/userRoutes.js]: Use arrow functions for code consistency.

File: senz-web/backend/routes/userRoutes.js [hunk 3]
```diff
@@ -70,7 +72,7 @@ router.post("/register", function(req, res) {
         password: hashedPass,
         signature: getSignature(String(req.body.name))
       },
-      function(err, user) {
+      function (err, user) {
```
Comment (YashMeh) [inline senz-web/backend/routes/userRoutes.js]: Add comments to make the code self-explanatory.

**Comment by YashMeh** [comment]
@carpecodeum add comments to the codebase to make it easier for other members to understand,
also make sure to add the steps to configure the nodemailer in the backend/README in your next PR.. :)

**Comment by carpecodeum** [comment]
> @carpecodeum add comments to the codebase to make it easier for other members to understand,
> also make sure to add the steps to configure the nodemailer in the backend/README in your next PR.. :)

okay sure!

**Pull request merged**